src/connectors/paymentService.ts

```diff
@@ -619,6 +619,28 @@ export class PaymentService extends BaseService {
     }
   }
 
+  isTransactionSplitted = async ({
+    providerTxId,
+    amount,
+  }: {
+    providerTxId: string
+    amount: number
+  }) => {
+    const splitTxs = await this.knex('transaction')
+      .select()
+      .where({
+        purpose: TRANSACTION_PURPOSE.subscriptionSplit,
+        state: TRANSACTION_STATE.succeeded,
+        remark: `stripe:${providerTxId}`,
```

Comment:
shall we add an index for this query?

Reply:
> shall we add an index for this query?

Or, shall we have a new field to refer to the parent transaction id?